src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -33,19 +38,22 @@ internal ItemManager(RecentItemsManager recentItemsManager, IGlobalOptionService
             CancellationToken cancellationToken)
         {
             var stopwatch = SharedStopwatch.StartNew();
-            var sessionData = CompletionSessionData.GetOrCreateSessionData(session);
-            // This method is called exactly once, so use the opportunity to set a baseline for telemetry.
-            if (sessionData.TargetTypeFilterExperimentEnabled)
-            {
-                AsyncCompletionLogger.LogSessionHasTargetTypeFilterEnabled();
-                if (data.InitialList.Any(static i => i.Filters.Any(static f => f.DisplayText == FeaturesResources.Target_type_matches)))
-                    AsyncCompletionLogger.LogSessionContainsTargetTypeFilter();
-            }
+            var items = SortCompletionitems(session, data).ToImmutableArray();
 
-            // Sort by default comparer of Roslyn CompletionItem
-            var sortedItems = data.InitialList.OrderBy(CompletionItemData.GetOrAddDummyRoslynItem).ToImmutableArray();
             AsyncCompletionLogger.LogItemManagerSortTicksDataPoint((int)stopwatch.Elapsed.TotalMilliseconds);
-            return Task.FromResult(sortedItems);
+            return Task.FromResult(items);
+        }
+
+        public Task<CompletionList<VSCompletionItem>> SortCompletionItemListAsync(
+            IAsyncCompletionSession session,
+            AsyncCompletionSessionInitialDataSnapshot data,
+            CancellationToken cancellationToken)
```

Comment:
looks like we lose the cancellation tokrn and don't take advantage of it when sorting.